File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let map = {};

        for ( let i = 0; i < nums.length; i++) {
            map[nums[i]] ? map[nums[i]]++ : map[nums[i]] = 1; 
        }

        let sortedMap = Object.entries(map).sort((a, b) => b[1] - a[1]);

        return sortedMap.slice(0,k).map(item => item[0]);
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums: return 0
        nums.sort()
        result = 1
        current_streak = 1
        i = 1
        while i < len(nums):
            if nums[i] != nums[i - 1]:
                if nums[i - 1] == nums[i] - 1:
                    current_streak += 1
                else:
                    result = max(result, current_streak)
                    current_streak = 1
            i += 1

        return max(result, current_streak)